### PR TITLE
Update steve for new project filtering feature

### PR DESCRIPTION
**Pull request by cmurphy**
https://github.com/rancher/rancher/issues/40140
Depends on https://github.com/rancher/steve/pull/105

This PR updates the steve dependency to use the new project filtering feature and adds integration tests. It also includes some changes to make the integration tests more stable and resilient.

### CODE REVIEW

File: tests/v2/integration/steveapi/steve_api_test.go [hunk 1]
```diff
@@ -1697,6 +2372,11 @@ func (s *steveAPITestSuite) TestList() {
 			}
 			query, err := url.ParseQuery(test.query)
 			require.NoError(s.T(), err)
+			if _, ok := query["sort"]; !ok && s.clusterID != "local" {
+				// k8s does not guarantee any particular order but usually returns results sorted by namespace and name.
+				// k3d seems to have its own ideas, so we can't rely on a consistent order when testing on the downstream cluster.
```
Comment (KevinJoiner) [inline tests/v2/integration/steveapi/steve_api_test.go:2377]: 😆

**Review by KevinJoiner — APPROVED** [review]

File: tests/v2/integration/steveapi/steve_api_test.go (hunk 1 above)
Comment (igomez06) [reply tests/v2/integration/steveapi/steve_api_test.go:2377]: I did notice the same thing on any downstream cluster :grin:

**Review by igomez06 — APPROVED** [review]
Looks great, and I know I sound like a broken record, I think we should port the import, and setup updates to the v2.6 branch at minimum. Up to you if you think some of Steve test updates should be there as well. Thank you!

**Comment by cmurphy** [comment]
@igomez06 will definitely port the test setup parts to 2.6

**Pull request merged**